### PR TITLE
Add support for projected volume kubeconfig

### PR DESCRIPTION
**Pull request by timuthy**
/area security
/kind enhancement

**What this PR does / why we need it**:
This PR adds the option to configure a kubeconfig projected volume for the dashboard deployment. It is for example needed, if operators generate their kubeconfigs for the virtual garden cluster via the [TokenRequestor](https://github.com/gardener/gardener/blob/master/docs/concepts/resource-manager.md#tokeninvalidator-controller).

**Release note**:

```feature operator
The `dashboard` chart allows to optionally configure a projected volume based kubeconfig.
```

### CODE REVIEW

File: charts/gardener-dashboard/values.yaml [hunk 1]
```diff
@@ -29,6 +29,11 @@ global:
       expirationSeconds: 43200 # 12 hours
       audience: ''
 
+  # projectedKubeconfig:
```
Comment (petersutter) [inline charts/gardener-dashboard/values.yaml:32]: ideally, also add a comment here describing this option

**Review by petersutter — APPROVED** [review]
just a nit regarding adding an additional comment in the values.yaml file to describe the option, otherwise: /lgtm

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Review by grolu — APPROVED** [review]
/lgtm

**Pull request merged**